mempool/blockchain.go

```diff
@@ -0,0 +1,158 @@
+package mempool
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/ethereum/go-ethereum/params"
+
+	"github.com/cosmos/evm/mempool/txpool"
+	"github.com/cosmos/evm/mempool/txpool/legacypool"
+	"github.com/cosmos/evm/x/vm/statedb"
+	evmtypes "github.com/cosmos/evm/x/vm/types"
+
+	errors2 "cosmossdk.io/errors"
+	types2 "cosmossdk.io/store/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var (
+	_ txpool.BlockChain     = Blockchain{}
+	_ legacypool.BlockChain = Blockchain{}
+)
+
+// Blockchain implements the BlockChain interface required by Ethereum transaction pools.
+// It bridges Cosmos SDK blockchain state with Ethereum's transaction pool system by providing
+// access to block headers, chain configuration, and state databases. This implementation is
+// specifically designed for instant finality chains where reorgs never occur.
+type Blockchain struct {
+	ctx                func(height int64, prove bool) (sdk.Context, error)
```

Comment:
func(height int64, prove bool) (sdk.Context, error)

should we define this function as a named type?

`type getCtxCallback = func(height int64, prove bool) (sdk.Context, error)`

Reply:
fixeroo: https://github.com/cosmos/evm/pull/387/commits/e01997724de0f20173603fb9725bdb80d808bc27